exchanges/okcoin/okcoin_websocket.go

```diff
@@ -299,468 +500,326 @@ func (o *OKCoin) wsProcessTrades(respRaw []byte) error {
 		if len(orderbookData.Bids)-1 >= i {
-			price := strconv.FormatFloat(orderbookData.Bids[i].Price, 'f', -1, 64)
-			amount := strconv.FormatFloat(orderbookData.Bids[i].Amount, 'f', -1, 64)
-			checksum.WriteString(price + delimiterColon + amount + delimiterColon)
+			bidPrice := strconv.FormatFloat(orderbookData.Bids[i].Price, 'f', -1, 64)
+			bidAmount := strconv.FormatFloat(orderbookData.Bids[i].Amount, 'f', -1, 64)
+			checksum.WriteString(bidPrice +
+				currency.ColonDelimiter +
+				bidAmount +
+				currency.ColonDelimiter)
 		}
 		if len(orderbookData.Asks)-1 >= i {
-			price := strconv.FormatFloat(orderbookData.Asks[i].Price, 'f', -1, 64)
-			amount := strconv.FormatFloat(orderbookData.Asks[i].Amount, 'f', -1, 64)
-			checksum.WriteString(price + delimiterColon + amount + delimiterColon)
+			askPrice := strconv.FormatFloat(orderbookData.Asks[i].Price, 'f', -1, 64)
+			askAmount := strconv.FormatFloat(orderbookData.Asks[i].Amount, 'f', -1, 64)
+			checksum.WriteString(askPrice +
+				currency.ColonDelimiter +
+				askAmount +
+				currency.ColonDelimiter)
 		}
 	}
-	checksumStr := strings.TrimSuffix(checksum.String(), delimiterColon)
+	checksumStr := strings.TrimSuffix(checksum.String(), currency.ColonDelimiter)
 	return int32(crc32.ChecksumIEEE([]byte(checksumStr)))
 }
 
 // GenerateDefaultSubscriptions Adds default subscriptions to websocket to be
 // handled by ManageSubscriptions()
-func (o *OKCoin) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
+func (o *Okcoin) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
 	var subscriptions []stream.ChannelSubscription
-	assets := o.GetAssetTypes(true)
-	for x := range assets {
-		pairs, err := o.GetEnabledPairs(assets[x])
-		if err != nil {
-			return nil, err
-		}
-
-		if assets[x] != asset.Spot {
-			o.Websocket.DataHandler <- fmt.Errorf("%w %v", asset.ErrNotSupported, assets[x])
-			continue
-		}
-		channels := defaultSpotSubscribedChannels
-		if o.IsWebsocketAuthenticationSupported() {
-			channels = append(channels,
-				okcoinWsSpotMarginAccount,
-				okcoinWsSpotAccount,
-				okcoinWsSpotOrder)
-		}
-		for i := range pairs {
-			p, err := o.FormatExchangeCurrency(pairs[i], asset.Spot)
-			if err != nil {
-				return nil, err
+	pairs, err := o.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		return nil, err
+	}
+	spotFormat, err := o.GetPairFormat(asset.Spot, true)
+	if err != nil {
+		return nil, err
+	}
+	pairs = pairs.Format(spotFormat)
+	channels := defaultSubscriptions
+	if o.Websocket.CanUseAuthenticatedEndpoints() {
+		channels = append(
+			channels,
+			wsAccount,
+			wsOrder,
+			wsOrdersAlgo,
+			wsAlgoAdvance,
+		)
+	}
+	for s := range channels {
+		switch channels[s] {
+		case wsInstruments:
+			subscriptions = append(subscriptions, stream.ChannelSubscription{
+				Channel: channels[s],
+				Asset:   asset.Spot,
+			})
+		case wsTickers, wsTrades, wsOrderbooks, wsOrderbooksL5, wsOrderbookL1, wsOrderbookTickByTickL50,
```

Comment:
Having switched the default orderbook to `wsOrderbookTickByTickL400`, I'm not getting orderbooks, please ensure the various endpoints work when subscribed

Reply:
Change not implemented because of the reason above.